File: ex-14/ex-14_java/VertexArray.java
public class VertexArray {
    public static final int LineStrip = 0;

    public static class Vertice {
        public Vetor2f position;
        public Cor color;

        public Vertice() {
            this.position = new Vetor2f(0, 0);
            this.color = new Cor(255, 255, 255);
        }
    }

    private int tipoPrimitiva;
    private Vertice[] vertices;

    public VertexArray(int tipoPrimitiva, int contagem) {
        this.tipoPrimitiva = tipoPrimitiva;
        this.vertices = new Vertice[contagem];
        for (int i = 0; i < contagem; i++) {
            this.vertices[i] = new Vertice();
        }
    }

    public Vertice get(int i) {
        return vertices[i];
    }

    public int getVertexCount() {
        return vertices.length;
    }

    public int getTipoPrimitiva() {
        return tipoPrimitiva;
    }
}

File: ex-14/ex-14_python/vertex_array.py
from cor import Cor
from vetor2f import Vetor2f

LineStrip = 0


class Vertice:
    def __init__(self):
        self.position = Vetor2f(0, 0)
        self.color = Cor(255, 255, 255)


class VertexArray:
    def __init__(self, tipoPrimitiva, contagem):
        self.tipoPrimitiva = tipoPrimitiva
        self.vertices = [Vertice() for _ in range(contagem)]

    def __getitem__(self, i):
        return self.vertices[i]

    def getVertexCount(self):
        return len(self.vertices)

    def getTipoPrimitiva(self):
        return self.tipoPrimitiva
